docs/src/examples/jump_umap_analysis_with_cosmicqc.py

```diff
@@ -0,0 +1,421 @@
+print("\n".join(str(schema).split("\n")[:12]))
+
+# +
+# set a list of metadata columns for use throughout
+metadata_cols = [
+    "Metadata_ImageNumber",
+    "Image_Metadata_Row",
+    "Image_Metadata_Site",
+    "Metadata_ObjectNumber",
+    "Metadata_Plate",
+    "Metadata_Well",
+    "Image_TableNumber",
+]
+
+# read only metadata columns with outlier-specific columns
+df_merged_single_cells = pd.read_parquet(
+    path=merged_single_cells,
+    columns=[
+        *metadata_cols,
+        "Nuclei_AreaShape_Area",
+        "Nuclei_AreaShape_FormFactor",
+        "Nuclei_AreaShape_Eccentricity",
+    ],
+)
+df_merged_single_cells.head()
+# -
+
+# label outliers within the dataset
+print("Large nuclei outliers:")
+df_labeled_outliers = cosmicqc.analyze.find_outliers(
+    df=df_merged_single_cells,
+    metadata_columns=metadata_cols,
+    feature_thresholds="large_nuclei",
+)
+
+# label outliers within the dataset
+print("Elongated nuclei outliers:")
+df_labeled_outliers = cosmicqc.analyze.find_outliers(
+    df=df_merged_single_cells,
+    metadata_columns=metadata_cols,
+    feature_thresholds="elongated_nuclei",
+)
+
+# label outliers within the dataset
+print("Small and low formfactor nuclei outliers:")
+df_labeled_outliers = cosmicqc.analyze.find_outliers(
+    df=df_merged_single_cells,
+    metadata_columns=metadata_cols,
+    feature_thresholds="small_and_low_formfactor_nuclei",
+)
+
+# label outliers within the dataset
+df_labeled_outliers = cosmicqc.analyze.label_outliers(
+    df=df_merged_single_cells,
+    include_threshold_scores=True,
+)
+# show added columns
+df_labeled_outliers[
+    [col for col in df_labeled_outliers.columns.tolist() if "cqc." in col]
+].head()
+
+# +
+# create a column which indicates at least one outlier being found
+df_labeled_outliers["analysis.included_at_least_one_outlier"] = df_labeled_outliers[
+    [col for col in df_labeled_outliers.columns.tolist() if ".is_outlier" in col]
+].any(axis=1)
+
+# show value counts for all outliers
+outliers_counts = df_labeled_outliers[
+    "analysis.included_at_least_one_outlier"
+].value_counts()
+outliers_counts
+# -
+
+# show the percentage of total dataset
+print(
+    (outliers_counts.iloc[1] / outliers_counts.iloc[0]) * 100,
+    "%",
+    "of",
+    outliers_counts.iloc[0],
+    "include erroneous outliers of some kind.",
+)
+
+# show histograms to help visualize the data
+df_labeled_outliers.show_report()
+# +
+parquet_sampled_with_outliers = "./BR00117012_sampled_with_outliers.parquet"
+
+# check if we already have normalized data
+if not pathlib.Path(parquet_sampled_with_outliers).is_file():
+    # set a fraction for sampling
+    sample_fraction = 0.44
+
+    # read the dataset
+    df_features = pd.read_parquet("./BR00117012.parquet")
+
+    # group by metadata_well for all features then sample
+    # the dataset by a fraction.
+    df_features = (
+        df_features.groupby(["Metadata_Well"])[df_features.columns]
```

Comment:
```suggestion
        df_features.groupby(["Metadata_Well"])
```

I don't think this is necessary

Reply:
Thanks for catching this - I agree it does look a bit unusual for syntax. I found I had to add the column selection to avoid a warning from Pandas regarding how groupby is used here. Going to retain this code for now. See here for more: https://stackoverflow.com/questions/77969964/deprecation-warning-with-groupby-apply

A warning in the output of the cell otherwise shows:

DeprecationWarning:
DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.